File: 프로그래머스/lv1/12915. 문자열 내 마음대로 정렬하기/문자열 내 마음대로 정렬하기.cpp
#include <bits/stdc++.h>
using namespace std;
int n;
bool cmp(string a, string b) {
    if(a[n] != b[n]) return a[n] < b[n];
    return a < b;
}
vector<string> solution(vector<string> strings, int n) {
    ::n = n;
    sort(strings.begin(), strings.end(), cmp);
    return strings;
}

// import java.util.*;
// class Solution {
//     public String[] solution(String[] strings, int n) {
//         Arrays.sort(strings, (s1, s2) 
//                     -> (s1.charAt(n) == s2.charAt(n)) ? s1 - s2 : s1.charAt(n) - s2.charAt(n));
//         return strings;
//     }
// }

File: 프로그래머스/lv1/12915. 문자열 내 마음대로 정렬하기/문자열 내 마음대로 정렬하기.java
import java.util.*;

class Solution {
    public String[] solution(String[] strings, int n) {
        Arrays.sort(strings, new Comparator<String>(){
           public int compare(String a, String b) {
               if(a.charAt(n) != b.charAt(n)) {
                   return a.charAt(n) - b.charAt(n);
               }
               return a.compareTo(b);
           }
        });
        return strings;
    }
}